### PR TITLE
Add teos10

### PR DESCRIPTION
**Pull request by mvdh7**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by mvdh7** [comment]
@conda-forge/help-python ready for review!

Fails on Windows due to jax dependency.

File: recipes/teos10/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  sha256: 391ab9c700617260d62980e500a313b970d3f8fbf3807cb2608ad4d08525c70d
+
+build:
+  noarch: python
+  script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
+  number: 0
+
+requirements:
+  host:
+    - python >={{ python_min }}
```
Comment (ocefpaf) [inline recipes/teos10/meta.yaml:20]:
```suggestion
    - python {{ python_min }}
```

File: recipes/teos10/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    - setuptools
+    - pip
+  run:
+    - python >={{ python_min }}
```
Comment (ocefpaf) [inline recipes/teos10/meta.yaml:24]:
```suggestion
    - python {{ python_min }}
```

File: recipes/teos10/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    - jax
+
+test:
+  imports:
+    - teos10
+  commands:
+    - pip check
+  requires:
+    - python >={{ python_min }}
```
Comment (ocefpaf) [inline recipes/teos10/meta.yaml:33]:
```suggestion
    - python {{ python_min }}
```

**Comment by ocefpaf** [comment]
Hi, author of the "official" gsw here. I'd love to see a test suite that compares the results of both. While your approach here is quite elegant, the official one is based on the literature on the topic, with many tests, careful precision adjustments, and multi-language implementation.

BTW, this is not related to your conda-forge review, just my personal curiosity. To get this in conda-forge all you need to do is to address those two lints I commented in the code rewview.

**Comment by mvdh7** [comment]
Hi @ocefpaf, thanks for your comments (and review), I think those tests are a great idea - I will add it to my list and come back to that when I have time (next year) & keep you updated.

For clarity, the reason I made this version is not with any intention to be a replacement the gsw package, but mainly because I need jax-compatible versions of these functions for a different project.

**Pull request merged**